operator/v1/types_ingress.go

```diff
@@ -1399,6 +1399,132 @@ type IngressControllerHTTPHeaders struct {
 	// +nullable
 	// +optional
 	HeaderNameCaseAdjustments []IngressControllerHTTPHeaderNameCaseAdjustment `json:"headerNameCaseAdjustments,omitempty"`
+
+	// actions specifies options for performing action on headers.
+	// Note that this option only applies to cleartext HTTP connections
+	// and to secure HTTP connections for which the ingress controller
+	// terminates encryption (that is, edge-terminated or reencrypt
+	// connections).  Headers cannot be set or deleted for TLS passthrough
+	// connections.
+	// Setting HSTS header is supported via another mechanism in Ingress.Spec.RequiredHSTSPolicies.
+	// The header settings made here can replace settings made in: cache-control, spec.clientTLS,
+	// spec.httpHeaders.forwardedHeaderPolicy, spec.httpHeaders.uniqueId,
+	// and spec.httpHeaders.headerNameCaseAdjustments.
```

Comment:
`can` is ambiguous, what is the behaviour if I set cache-control or one of these other options, I assume these actions override anything set there?

In which case we should make that explicit.
```
Any actions defined here will take precedence over the following other fields; ...
```

The names you have for these fields also differ in format, some appear to be fully qualified while others are not, please be consistent

Reply:
How about, "Any actions defined here are applied after any actions related to the following other fields"?  For example, if you specify a header using both `spec.httpHeaders.uniqueId` and `spec.httpHeaders.actions`, I believe the `uniqueId` header does get set, and then the action from `actions` can read, overwrite, or delete the header.  Miheer, correct me if I'm wrong.

Reply:
Yeah that works for me if you think that is more accurately describing the behaviour

Reply:
Yes @Miciah